iml-agent/src/action_plugins/check_stonith.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::{CibError, ImlAgentError},
+    cmd::cmd_output,
+};
+use elementtree::Element;
+use futures::Future;
+use tracing::{span, Level};
+use tracing_futures::Instrument;
+
+fn stonith_ok(elem: &Element) -> Result<(bool, String), ImlAgentError> {
+    match elem.get_attr("type") {
+        None => Err(ImlAgentError::CibError(CibError(format!(
+            "{} is missing type attribute",
+            elem.tag()
+        )))),
+        Some("fence_chroma") => match elem.find("instance_attributes") {
+            None => Ok((false, "fence_chroma - unconfigured".to_string())),
+            Some(_) => Ok((true, "fence_chroma".to_string())),
+        },
+        Some(t) => Ok((true, t.to_string())),
+        // @@TODO: This should check for node name in
+        // instance_attributes:pcmk_host_list if
+        // pcmk_host_check=static-list
+
+        // @@TODO: This should check meta_attributes:target-role is
+        // not Stopped (disabled agent)
+    }
+}
+
+pub fn check_stonith(_: ()) -> impl Future<Item = (bool, String), Error = ImlAgentError> {
```

Comment:
This looks fine to me but one thing we may want to consider is to move the `cmd_output` outside of this function and instead let it take the future that is returned. This would allow us to write a unit test with a sample cibadmin output.